fjord/feedback/static/js/android_feedback.js

```diff
@@ -1,62 +1,51 @@
-(function($) {
-  'use strict';
-
-  // Settings
-  var browserReviewUrl = 'https://play.google.com/store/apps/details?id=org.mozilla.CHANNEL#details-reviews';
-  var deviceReviewUrl = 'market://details?id=org.mozilla.CHANNEL';
-
-  // Initialize
-  var reviewUrl = '';
-  var lastUrl = '';
-  var url = decodeURI(window.location.href);
-
-  /*
-    Decide whether they're here deliberately or due to a feedback prompt
-    (Fx Android's feedback prompt includes a utm param);
-  */
-  var fromDevice = /source=feedback-prompt/.test(url);
-  if (fromDevice) {
-    reviewUrl = deviceReviewUrl;
-    /* If client prompted for feedback, delegate some event handling to the client */
-    $(document).on('click', '#maybe-later, #no-thanks', function(e) {
-        e.preventDefault();
-        var eventName = this.getAttribute('data-event');
-        var feedbackEvent = new Event(eventName, {"bubbles":true, "cancelable":false});
-        this.dispatchEvent(feedbackEvent);
+(function ($) {
+    'use strict';
+
+    // Settings
+    var browserReviewUrl = 'https://play.google.com/store/apps/details?id=org.mozilla.CHANNEL#details-reviews';
+    var deviceReviewUrl = 'market://details?id=org.mozilla.CHANNEL';
+
+    // Initialize
+    var reviewUrl = browserReviewUrl;
+    var channel = $('body').attr('data-channel');
+
+    /*
+      Fx Android fires an event when loaded by feedback prompt;
+      in that case act like we're on a device.
+
+      Simulate this in the console: Services.obs.notifyObservers(null, "Feedback:Show", null);
+    */
+    document.addEventListener('FeedbackPrompted', function(event) {
+        reviewUrl = deviceReviewUrl;
+        $('a.maybe-later, a.no-thanks').show();
+
+        // If client prompted for feedback, delegate some event handling to the client
+        $(document).on('click', 'a.maybe-later, a.no-thanks', function(e) {
+            e.preventDefault();
+            var eventName = this.getAttribute('data-event');
+            var feedbackEvent = new Event(eventName, {'bubbles':true, 'cancelable':false});
+            this.dispatchEvent(feedbackEvent);
+        });
+    });
+
+    $(document).on('click', '#happy-button-android', function(e) {
+      // Play store only has beta/release
+        if (channel != '' && /^(beta|release).*$/.test(channel)) {
+            if (channel == 'beta') {
+                reviewUrl = reviewUrl.replace('CHANNEL', 'firefox_beta');
+            }
+            else {
+                reviewUrl = reviewUrl.replace('CHANNEL', 'firefox');
+            }
+            $('#play-store').attr('href', reviewUrl);
+        }
+        else {
+            // If not beta/release, hide CTA to review on play store
+            $('.play-store-cta').hide();
+        }
+
+        // Trigger input's generic_feeback.js handler
+        $('#happy-button').click();
```

Comment:
My main complaint is that this obfuscates what code is actually running. If I'm just looking at this file, in order to figure out what code is called when this line runs, I need to look through the _entire_ codebase to find every click handler that could possibly be registered to `#happy-button`. That might include more than just stuff registered by ID; if the tag is an anchor, a handler for all `<a>` tags might also run.

Also, this requires that you include a hidden tag in the dom just to trigger this. The core thing you want is to reuse a piece of code in two places, and factoring that code into a function is the standard and clearest way to do this. Introduce the DOM and events just obscures things even if it happens to result in less lines changed.